src/mixins/allowable-actions-mixin.ts

```diff
@@ -2,101 +2,158 @@ import Vue from 'vue'
+        if (this.isFirm) {
+          return this.isAllowedFiling(FilingTypes.CHANGE_OF_REGISTRATION)
+        }
+        return this.isAllowedFiling(FilingTypes.CHANGE_OF_DIRECTORS)
       }
 
-      case AllowableActions.EDIT_BUSINESS_PROFILE: {
-        return !!businessId // FUTURE: disable for staff? see #14314
+      case AllowableActions.FREEZE_UNFREEZE: {
+        return this.isAllowedFiling(FilingTypes.ADMIN_FREEZE)
       }
 
-      case AllowableActions.FILE_ADDRESS_CHANGE: {
-        return (!this.isHistorical && !this.hasBlocker && !!businessId)
+      case AllowableActions.LIMITED_RESTO_CONVERT: {
+        return this.isAllowedFiling(FilingTypes.RESTORATION, FilingSubTypes.RESTORATION_CONVERSION)
       }
 
-      case AllowableActions.FILE_ANNUAL_REPORT: {
-        return (!this.isHistorical && !this.hasBlocker && !!businessId)
+      case AllowableActions.LIMITED_RESTO_EXTEND: {
+        return this.isAllowedFiling(FilingTypes.RESTORATION, FilingSubTypes.RESTORATION_EXTENSION)
       }
 
-      case AllowableActions.FILE_CORRECTION: {
-        return (!this.isHistorical && !this.hasBlocker && !!businessId && this.isRoleStaff)
+      case AllowableActions.PUT_BACK_ON: {
+        return this.isAllowedFiling(FilingTypes.PUT_BACK_ON)
       }
 
-      case AllowableActions.FILE_DIRECTOR_CHANGE: {
-        return (!this.isHistorical && !this.hasBlocker && !!businessId)
+      case AllowableActions.RECORD_CONVERSION: {
+        return this.isAllowedFiling(FilingTypes.CONVERSION)
       }
 
-      case AllowableActions.FILE_STAFF_NOTATION: {
-        return (!this.isHistorical && !this.hasBlockerExceptStaffApproval && !!businessId && this.isRoleStaff)
+      case AllowableActions.REGISTRARS_NOTATION: {
+        return this.isAllowedFiling(FilingTypes.REGISTRARS_NOTATION)
       }
 
-      case AllowableActions.VIEW_CHANGE_COMPANY_INFO: {
-        const isAllowedEntityType = (
-          this.isBenBcCccUlc ||
-          (this.isCoop && !!getFeatureFlag('special-resolution-ui-enabled')) ||
-          this.isPartnership ||
-          this.isSoleProp
+      case AllowableActions.REGISTRARS_ORDER: {
+        return this.isAllowedFiling(FilingTypes.REGISTRARS_ORDER)
+      }
+
+      case AllowableActions.RESTORATION: {
+        return (
+          this.isAllowedFiling(FilingTypes.RESTORATION, FilingSubTypes.RESTORATION_FULL) ||
+          this.isAllowedFiling(FilingTypes.RESTORATION, FilingSubTypes.RESTORATION_LIMITED)
         )
-        return (!this.isHistorical && !!businessId && isAllowedEntityType)
       }
 
-      case AllowableActions.VIEW_ADD_DIGITAL_CREDENTIALS: {
-        // Pilot is targeting specific Benefit Companies: Handled on LD Server Side
-        const isFeatureEnabled = !!getFeatureFlag('enable-digital-credentials')
-        const isNotaDcRoute = !(this.$route.matched.some(route => route.name === Routes.DIGITAL_CREDENTIALS))
+      case AllowableActions.STAFF_COMMENT: {
+        return (isBusiness && this.isRoleStaff)
```

Comment:
Eventually, this and maybe some other cases, will also come from the Legal API. For now this is fine.